### PR TITLE
Local storage

### PR DESCRIPTION
**Pull request by varanthir**
Save/Load Data in localStorage is done.
JSON used to put whole object in one record. Date is storred as (integer) miliseconds, not as date string.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -2,24 +2,31 @@
...
 
 //STORAGE
+function SaveData(notesList) {
+    localStorage.clear();
+    notesList.forEach((x, i) => {
+        localStorage.setItem(`n${i}`,
+            JSON.stringify(x)
+        )
+    });
+}
```
Comment (ulfryk) [inline script.js]: why not just:

```javascript
localStorage.set('varanthirToDo', JSON.stringify(notesList));
```

?

File: script.js [hunk 2]
```diff
@@ -2,24 +2,31 @@
...
     this.note = note;
-    this.date = date;
+    this.dateMiliSecs = date; //date store as (integer) miliseconds
     this.hour = hour; //bool, is hour setted?
 }
 
 //STORAGE
+function SaveData(notesList) {
+    localStorage.clear();
+    notesList.forEach((x, i) => {
```
Comment (ulfryk) [inline script.js]: **WTF** is `x` _eXtreme_ ? _Xena_ ? did you mean `(note, index) => …` ?

File: script.js [hunk 3]
```diff
@@ -2,24 +2,31 @@
...
-    notesList.push(new Note('Próba', 'Dyplom Aski (Vivaldi)', new Date(), false));
-    notesList.push(new Note('', 'Lekcja z Kr.', new Date(), true));
-    notesList.push(new Note('Dentysta', 'Rwanie 8ek..', new Date(), true));
-    notesList.push(new Note('Koncert', 'Gramy z Michaelem Jacksonem', new Date(), true));
-    notesList.push(new Note('Obiad', 'u babci.', new Date(), true));
-    notesList.push(new Note('Herbatka', 'u babci.', new Date(), true));
+    for(let i = 0; i < localStorage.length; i++) {
+        notesList.push(JSON.parse(localStorage.getItem(`n${i}`)));
+        //console.log(JSON.parse(localStorage.getItem(`n${i}`)));
+    }
```
Comment (ulfryk) [inline script.js]: same - just `localStorage.get('varanthirToDo')` would do the job

File: script.js [hunk 4]
```diff
@@ -2,24 +2,31 @@
...
 }
 
 //HELPERS
 function getDayOfWeek(date) {
     let d = ['Nd', 'Pn', 'Wt', 'Śr', 'Cz', 'Pt', 'Sb'];
-    //czy var w funkcji jest public?
+    //^czy var w funkcji jest public?
```
Comment (ulfryk) [inline script.js]: `var` has `function` scope (so it's not seen outside function it's declared in) while `let` and `const` both have {BLOCK} scope. So they can not be see even outside `if {…}` statement they are defined in.

It's good to not use `var` at all. Also using `let` is lame. Actually `const` can be used, but only if there is no other way. :)

File: script.js [hunk 5]
```diff
@@ -32,11 +39,12 @@ function getSimpleDate(date, hour) {
 } //jak łamać powyższe?
 
 //GUI
-function ShowNotes(htmlObj) {
+function ShowNotes(htmlObj, notesList) {
     let code = '';
-    MYAPP.notes.forEach((x) => {
+    notesList.forEach((x) => {
```
Comment (ulfryk) [inline script.js]: x? xXx? XxX? x?

File: script.js [hunk 6]
```diff
@@ -32,11 +39,12 @@ function getSimpleDate(date, hour) {
...
+        console.log(x.dateMiliSecs);
         code +=
-            `<p class="date">${getSimpleDate(x.date, x.hour)}</p>` +
+            `<p class="date">${getSimpleDate(new Date(x.dateMiliSecs), x.hour)}</p>` +
```
Comment (ulfryk) [inline script.js]: why do you `forEach` and reassign ?

File: index.html [hunk 7]
```diff
@@ -7,6 +7,8 @@
     </head>
 
     <body>
+        <button onclick="AddExtraNotes()">Add extra notes</button><!--delete when Add/Delete note is done-->
+        <button onclick="localStorage.clear()">Clear local storage</button><!--delete when Add/Delete note is done-->
```
Comment (ulfryk) [inline index.html]: It is a bad practice to define global functions like `AddExtraNotes`, it is not good to start function that is not a constructor with a Capital letter.

File: script.js [hunk 8]
```diff
@@ -2,24 +2,31 @@
 function Note(title, note, date, hour) {
     this.title = title; //can be ''
     this.note = note;
-    this.date = date;
+    this.dateMiliSecs = date; //date store as (integer) miliseconds
     this.hour = hour; //bool, is hour setted?
 }
 
 //STORAGE
+function SaveData(notesList) {
```
Comment (ulfryk) [inline script.js]: `saveData` it is not `C#` ;p

File: script.js [hunk 9]
```diff
@@ -2,24 +2,31 @@
...
+}
+
 function LoadData(notesList) {
-    notesList.push(new Note('Próba', 'Dyplom Aski (Vivaldi)', new Date(), false));
-    notesList.push(new Note('', 'Lekcja z Kr.', new Date(), true));
-    notesList.push(new Note('Dentysta', 'Rwanie 8ek..', new Date(), true));
-    notesList.push(new Note('Koncert', 'Gramy z Michaelem Jacksonem', new Date(), true));
-    notesList.push(new Note('Obiad', 'u babci.', new Date(), true));
-    notesList.push(new Note('Herbatka', 'u babci.', new Date(), true));
+    for(let i = 0; i < localStorage.length; i++) {
```
Comment (ulfryk) [inline script.js]: `loadData`

File: script.js [hunk 10]
```diff
@@ -32,11 +39,12 @@ function getSimpleDate(date, hour) {
 } //jak łamać powyższe?
 
 //GUI
-function ShowNotes(htmlObj) {
+function ShowNotes(htmlObj, notesList) {
```
Comment (ulfryk) [inline script.js]: `showNotes`

**Comment by ulfryk** [comment]
Please fix those few issues ;)

File: script.js (hunk 1 above)
Comment (varanthir) [inline script.js]: Thx, simple and wise. Done.

File: script.js (hunk 2 above)
Comment (varanthir) [inline script.js]: Xena is no more.

File: script.js (hunk 4 above)
Comment (varanthir) [inline script.js]: So it's generally good to not use vars and lets?

File: script.js (hunk 3 above)
Comment (varanthir) [inline script.js]: Done.

File: script.js (hunk 5 above)
Comment (varanthir) [inline script.js]: Done.

File: index.html (hunk 7 above)
Comment (varanthir) [inline index.html]: Ok, it's developer's helper until adding new note by user will be done.

File: script.js (hunk 8 above)
Comment (varanthir) [inline script.js]: Done.

File: script.js (hunk 9 above)
Comment (varanthir) [inline script.js]: Done.

File: script.js (hunk 10 above)
Comment (varanthir) [inline script.js]: Done.

File: script.js (hunk 6 above)
Comment (varanthir) [inline script.js]: What would be better?

Comment (ulfryk) [inline script.js]: if you have to accumulate a value from items of some collection - use `.reduce(…)`

just:

```javascript
htmlObj.innerHTML = notesList.reduce(…, '');
```

Comment (ulfryk) [inline script.js]: …or `.map(…).join('')` - it's quite good for concatenating lists…

but anyway - manual HTML creation is not a good idea - better use some tamplating lib (but not in this PR)

Comment (varanthir) [inline script.js]: Awesome

File: script.js [hunk 11]
```diff
@@ -2,25 +2,24 @@
 function Note(title, note, date, hour) {
     this.title = title; //can be ''
     this.note = note;
-    this.date = date;
+    this.dateMiliSecs = date; //date store as (integer) miliseconds
```
Comment (ulfryk) [inline script.js]: I somehow forgot to writ this -> `timestamp` is proper name for date stored as an integer in ms. Every developer will understand it ;)

**Comment by ulfryk** [comment]
One last issue ;)

File: script.js (hunk 11 above)
Comment (varanthir) [inline script.js]: Thx, me didn' know :)

**Comment by ulfryk** [comment]
Looks good :+1: - feel free to merge ;)

**Pull request merged**